src/main/java/com/thealgorithms/dynamicprogramming/SumOfSubset.java

```diff
@@ -2,17 +2,28 @@
 
 public class SumOfSubset {
 
-    public static boolean subsetSum(int[] arr, int num, int Key) {
-        if (Key == 0) {
-            return true;
-        }
-        if (num < 0 || Key < 0) {
-            return false;
-        }
+    static boolean subsetSum(int[] arr, int n, int Key) {
+
+        boolean[] prev = new boolean[Key + 1];
 
-        boolean include = subsetSum(arr, num - 1, Key - arr[num]);
-        boolean exclude = subsetSum(arr, num - 1, Key);
+        prev[0] = true;
 
-        return include || exclude;
+        if (arr[0] <= Key) {
+            prev[arr[0]] = true;
+        }
+        for (int ind = 1; ind <= n; ind++) {
+            boolean[] cur = new boolean[Key + 1];
+            cur[0] = true;
+            for (int target = 1; target <= Key; target++) {
+                boolean notTaken = prev[target];
+                boolean taken = false;
+                if (arr[ind] <= target) {
+                    taken = prev[target - arr[ind]];
+                }
+                cur[target] = notTaken || taken;
```

Comment:
(Just a remark) There could be some _micro optimizations_ done here by relying on _short-circuiting_.